suites/basic/suite.go

```diff
@@ -0,0 +1,168 @@
+	nsm.Suite
+}
+
+// TestLocalConnection deploys nsc, nse on the same node and checks that ping is working
+func (s *Suite) TestLocalConnection() {
+	out, err := s.Run(`kubectl get nodes -o go-template='{{range .items}}{{ if not .spec.taints  }}{{index .metadata.labels \"kubernetes.io/hostname\"}} {{end}}{{end}}'`)
+	s.Require().NoError(err)
+	nodes := strings.Split(strings.TrimSpace(out), " ")
+	s.Require().Greater(len(nodes), 1)
+
+	// Add kustomization
+
+	_, err = s.Runf(`
+cat << EOF > kustomization.yaml
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+bases:
+- ../examples/kernel-nsc
+- ../examples/kernel-nse
+
+namespace: %v
+
+patchesStrategicMerge:
+- patch-nsc.yaml
+- patch-nse.yaml
+EOF`, s.Namespace())
+
+	s.Require().NoError(err)
+
+	// Add patches
+	_, err = s.Runf(`
+cat << EOF > patch-nsc.yaml
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nsc
+spec:
+  nodeSelector: 
+    kubernetes.io/hostname: %v
+EOF`, nodes[0])
+
+	s.Require().NoError(err)
+
+	_, err = s.Runf(`
+cat << EOF > patch-nse.yaml
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nse
+spec:
+  nodeSelector: 
+    kubernetes.io/hostname: %v
+EOF`, nodes[0])
+
+	s.Require().NoError(err)
+
+	_, err = s.Run(`kubectl apply -k .`)
+	s.Require().NoError(err)
+
+	_, err = s.Run("kubectl wait --for=condition=ready --timeout=15s pod -l app=nse -n " + s.Namespace())
+	s.Require().NoError(err)
+	_, err = s.Run("kubectl wait --for=condition=ready --timeout=15s pod -l app=nsc -n " + s.Namespace())
+	s.Require().NoError(err)
+
+	// TODO: Replace this to ping
+	_, err = s.Runf("kubectl logs nsc -n %v | grep \"All client init operations are done.\"", s.Namespace())
+	s.Require().NoError(err)
+}
+
+// TestRemoteConnection deploys nsc, nse on the different nodes and checks that ping is working
+func (s *Suite) TestRemoteConnection() {
+	out, err := s.Run(`kubectl get nodes -o go-template='{{range .items}}{{ if not .spec.taints  }}{{index .metadata.labels \"kubernetes.io/hostname\"}} {{end}}{{end}}'`)
+	s.Require().NoError(err)
+	nodes := strings.Split(strings.TrimSpace(out), " ")
+	s.Require().Greater(len(nodes), 1)
+
+	// Add kustomization
+
+	_, err = s.Runf(`
+cat << EOF > kustomization.yaml
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+bases:
+- ../examples/kernel-nsc
+- ../examples/kernel-nse
+
+namespace: %v
+
+patchesStrategicMerge:
+- patch-nsc.yaml
+- patch-nse.yaml
+EOF`, s.Namespace())
+
+	s.Require().NoError(err)
+	// Add patches
+	_, err = s.Runf(`
+cat << EOF > patch-nsc.yaml
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nsc
+spec:
+  nodeSelector:
+    kubernetes.io/hostname: %v
+EOF`, nodes[0])
+
+	s.Require().NoError(err)
+
+	_, err = s.Runf(`
+cat << EOF > patch-nse.yaml
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nse
+spec:
+  nodeSelector:
+    kubernetes.io/hostname: %v
+EOF`, nodes[1])
+
+	s.Require().NoError(err)
+
+	_, err = s.Run(`kubectl apply -k .`)
+	s.Require().NoError(err)
```

Comment:
@denis-tingajkin  Let's move s.Require().NoError(err) into Run function internals. I think it will make tests a bit more clean.

Also Could we do a bit more and have:

```suggestion
	s.Run(`
	     export NAMESPACE=nsm # Runner will ignore default env variable.
	     kubectl apply -k .
	     kubectl wait --for=condition=ready --timeout=15s pod -l app=nse -n $(NAMESPACE)
	     kubectl wait --for=condition=ready --timeout=15s pod -l app=nsc -n $(NAMESPACE)
	     kubectl logs nsc -n $(NAMESPACE) | grep "All client init operations are done."
	`, map[string]string{'NAMESPACE': s.Namespace()}) // <- Runner will override some of variables
```

Reply:
@haiodo , @edwarnicke I feel the best solution for this could be https://github.com/networkservicemesh/integration-tests/issues/4